trie-db/src/iterator.rs

```diff
@@ -450,8 +473,15 @@ impl<L: TrieLayout> TrieDBRawIterator<L> {
 	/// Fetches the next trie item.
 	///
 	/// Must be called with the same `db` as when the iterator was created.
-	pub fn next_item(&mut self, db: &TrieDB<L>) -> Option<TrieItem<TrieHash<L>, CError<L>>> {
-		while let Some(raw_item) = self.next_raw_item(db) {
+	///
+	/// Specify `fwd` to indicate the direction of the iteration (`true` for forward, `false` for
+	/// backward).
+	pub fn next_item(
+		&mut self,
+		db: &TrieDB<L>,
+		fwd: bool,
```

Comment:
So this change a "not so public" api (raw db iterator should not be use everywhere), so should be remember this require correct versioning changes when publishing.

Reply:
(could also make a new function `prev_item`, but in this case an internal function with the parameter seems still needed to avoid code redundancy).

Reply:
> So this change a "not so public" api (raw db iterator should not be use everywhere), so should be remember this require correct versioning changes when publishing.

You mean it's public but not many people should be using it? We would still need to bump the version to the correct one to convey it's breaking the api. 

> (could also make a new function `prev_item`, but in this case an internal function with the parameter seems still needed to avoid code redundancy).

I initially did this, but I found it much cleaner to just have `next_item`. I could also extract the core logic of the function into a reusable one so that there is no redundant code.